app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v3.rb

```diff
@@ -227,9 +227,24 @@ def method_missing(method_name, *args, &block)
       end
     end
 
-    def find_mac_address_on_network(nics, network, log)
+    def get_mac_address_of_nic_on_requested_vlan(args)
+      network = find_network_in_cluster(args[:value_of_vlan_option], args[:dest_cluster], args[:source_template])
+      return nil if network.nil?
+
+      find_mac_address_on_network(network, args[:destination])
+    end
+
+    def find_network_in_cluster(network_name, dest_cluster, source)
+      network = source.ext_management_system.ovirt_services.cluster_find_network_by_name(dest_cluster.ems_ref, network_name)
```

Comment:
1. Can't you solve it by invoking the code (in network.rb) using the template's ems? (If the source ems is not initialized yet, it seems to be the correct approach).
2. How 'nics_for_vm(..)' which is called directly from 'find_mac_address_on_network(..)' is working?
Doesn't it need the ems as well to get the connection?